gateway-js/src/index.ts

```diff
@@ -346,6 +359,43 @@ export class ApolloGateway implements GraphQLService {
     }
   }
 
+  private issueDeprecationWarningsIfApplicable() {
+    // TODO(trevor:removeServiceList)
+    if ('experimental_updateSupergraphSdl' in this.config) {
+      this.logger.warn(
+        'The `experimental_updateSupergraphSdl` option is deprecated and will be removed in a future version of `@apollo/gateway`. Please migrate to the function form of the `supergraphSdl` configuration option.',
+      );
+    }
+
+    // TODO(trevor:removeServiceList)
+    if ('experimental_updateServiceDefinitions' in this.config) {
+      this.logger.warn(
+        'The `experimental_updateServiceDefinitions` option is deprecated and will be removed in a future version of `@apollo/gateway`. Please migrate to the function form of the `supergraphSdl` configuration option.',
+      );
+    }
+
+    // TODO(trevor:removeServiceList)
+    if ('serviceList' in this.config) {
+      this.logger.warn(
+        'The `serviceList` option is deprecated and will be removed in a future version of `@apollo/gateway`. Please migrate to the function form of the `supergraphSdl` configuration option.',
+      );
+    }
+
+    // TODO(trevor:removeServiceList)
+    if ('localServiceList' in this.config) {
+      this.logger.warn(
+        'The `localServiceList` option is deprecated and will be removed in a future version of `@apollo/gateway`. Please migrate to the function form of the `supergraphSdl` configuration option.',
+      );
+    }
+
+    // TODO(trevor:removeServiceList)
+    if ('buildService' in this.config) {
```

Comment:
huh? how does supergraphSql replace buildService?

Reply:
Early mistake that I hadn't corrected yet
6482081f